### PR TITLE
flatten and flatMap tests

### PR DESCRIPTION
**Pull request by Shilpi3**

### CODE REVIEW

**Comment by michaelficarra** [comment]
I've pushed up a commit to clean some things up. I do think we need a few more basic positive tests, and agree with @littledan that we need to use more non-arrays that are `IsConcatSpreadable` (`arguments` objects) and `Iterable` (strings) as the *contents* of the flattened/flatMapped arrays, not just as the targets.

**Comment by rwaldron** [comment]
@michaelficarra your suggested additions can be done in follow up PRs, right?

File: test/built-ins/Array/prototype/flatMap/thisArg-argument.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-array.prototype.flatMap
+description: >
+    Behavior when thisArg is provided
+    Array.prototype.flatMap ( mapperFunction [ , thisArg ] )
+includes: [compareArray.js]
+---*/
+
+"use strict";
```
Comment (rwaldron) [inline test/built-ins/Array/prototype/flatMap/thisArg-argument.js]: Remove this line. If this test must only be executed in strict mode, then add the appropriate frontmatter: `flags: [onlyStrict]`

**Comment by littledan** [comment]
Btw a new feature flag wouldn't hurt either, though these are easy to filter anyway as they are in one directory.

**Comment by rwaldron** [comment]
@littledan yes, we need `Array.prototype.flatMap` and `Array.prototype.flatten` added to `features.txt` and added into all of these test files. I figured I could just add them after merging the tests.

**Comment by michaelficarra** [comment]
I've pushed another few commits addressing the feedback, including adding two new feature flags.

@rwaldron Yes, those additions can be done in follow-up PRs if we want implementations to be able to start using these.

**Comment by rwaldron** [comment]
> including adding two new feature flags.

I appreciate that, thank you. 

> Yes, those additions can be done in follow-up PRs if we want implementations to be able to start using these.

👍

**Pull request merged**